contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "hardhat/console.sol";
+import "./Ownable.sol";
+
+contract DomainRegistry is Ownable {
+    uint256 public totalDomains;
+    uint256 public totalReservedDomains;
+
+    struct Domain {
+        address controller;
+        uint256 deposit;
+    }
+
+    mapping(string => Domain) private domains;
+
+    event DomainReserved(string indexed domainName, address indexed controller, uint256 deposit);
+    event DepositChanged(string indexed domainName, uint256 newDeposit);
+    event DomainControlTransferred(string indexed domainName, address indexed newController);
+    event DomainReleased(string indexed domainName, address indexed controller, uint256 refundAmount);
+
+    constructor() {}
+
+    function reserveDomain(string memory domainName, uint256 initialDeposit) public payable {
+        require(msg.value >= initialDeposit, "Insufficient initial deposit");
+        require(domains[domainName].controller == address(0), "Domain already reserved");
+        require(isTopLevelDomain(domainName), "Domain name must be a top-level domain");
+
+        domains[domainName] = Domain({
+            controller: msg.sender,
+            deposit: initialDeposit
+        });
+        totalDomains++;
+        totalReservedDomains++;
+        
+        emit DomainReserved(domainName, msg.sender, initialDeposit);
+    }
+
+    function isTopLevelDomain(string memory domainName) private pure returns (bool) {
+    bytes memory domainBytes = bytes(domainName);
+    for(uint i = 0; i < domainBytes.length; i++) {
+        if(domainBytes[i] == '.') {
+            return false;
+        }
+    }
+    return true;
+    }
+
+    function changeDeposit(string memory domainName, uint256 newDeposit) public payable {
+        require(domains[domainName].controller == msg.sender, "Only the domain controller can change deposit");
+        require(msg.value >= newDeposit, "Insufficient deposit amount");
```

Comment:
Why not to introduce modifiers (everywhere) to increase the code readability? Besides you repeat the same code many times.
-1 point